src/Connection/FiniteConnectionPool.php

```diff
@@ -41,10 +41,23 @@ final class FiniteConnectionPool implements ConnectionPool
     /** @var int */
     private $openConnectionCount = 0;
 
-    public function __construct(int $maxConnections, ?ConnectionFactory $connectionFactory = null)
+    /**
+     * Create a connection pool that limits the number of connections per authority to $maxConnections.
+     *
+     * @param int                    $maxConnections
+     * @param ConnectionFactory|null $connectionFactory
+     *
+     * @return self
+     */
+    public static function perAuthority(int $maxConnections, ?ConnectionFactory $connectionFactory = null): self
```

Comment:
byHost for parity with LimitedConnectionPool::byHost? or byAuthority?

Reply:
`byAuthority` would probably be best, since it is different than `LimitedConnectionPool::byHost`, as `FiniteConnectionPool::byAuthority(1)` would allow a connection to `http://example.com:80` and `https://example.com:443`.